File: Algorithms/DifferentAddends/DifferentAddends.py
def get_different_addends_number(input_number):
    assert input_number >= 1

    addends_number = 1
    addends_sum = 0
    addends = []

    while addends_sum <= input_number:
        if input_number <= 2:
            addends.append(input_number)
            return 1, addends
        
        if addends_sum + addends_number <= input_number:
            addends.append(addends_number)
            addends_sum += addends_number
            addends_number += 1
        else:
            addends_number -= 1

            while addends_sum <= input_number:
                if input_number == addends_sum:
                    return addends_number, addends

                removed_element = addends[-1]
                addends.pop()
                addends.append(removed_element + 1)
                addends_sum += 1

            return addends_number, addends


def main():
    input_number = int(input())
    addends = []
    addends_number, addends = get_different_addends_number(input_number)
    print(addends_number)

    for idx, _ in enumerate(addends):
        if idx < len(addends) - 1:
            print(addends[idx], end=' ')
        else:
            print(addends[idx])


if __name__ == '__main__':
    main()

File: Algorithms/DifferentAddends/DifferentAddends.cs
using System.Diagnostics;


namespace DifferentAddends
{
    public static class DifferentAddends
    {
        public static (int, List<int>) GetDifferentAddendsNumber(this int number)
        {
            Debug.Assert(number >= 1);

            int addendsNumber = 1;
            int addendsSum = 0;
            List<int> addends = new();

            while(addendsSum <= number)
            {
                if (number <= 2)
                {
                    addends.Add(addendsNumber);
                    return (1, addends);
                }

                if (addendsSum + addendsNumber <= number)
                {
                    addends.Add(addendsNumber);
                    addendsSum += addendsNumber;
                    addendsNumber += 1;
                }
                else
                {
                    addendsNumber--;
                    while (addendsSum <= number)
                    {
                        if (number == addendsSum)
                        {
                            return (addendsNumber, addends);
                        }

                        int removedElement = addends.Last();
                        addends.RemoveAt(addends.Count - 1);
                        addends.Add(removedElement + 1);
                        addendsSum += 1;
                    }

                    return (addendsNumber, addends);
                }
            }

            return (addendsNumber, addends);
        }

        public static void Main()
        {
            int number = int.Parse(Console.ReadLine());

            (int addendsNumber, List<int> addends) = number.GetDifferentAddendsNumber();

            Console.WriteLine(addendsNumber);
            addends.ForEach(x => Console.Write("{0} ", x));
        }
    }
}